### DISCUSSION TITLE
My simple experimental test rig

### DISCUSSION THREAD

**Discussion by stefan63** [DIY-Flow-Bench/DIY-Flow-Bench#174] [Build Threads]
My ruff not particulary well designed test rig using 75mm YD 70mm ID drain tubes.
Having dual vacuum cleaner motors at a bout 1400W a piece I also have a 5KW speed regulator that works just fine.
Printed a MAF sensor adapter for the 75mm tube (Modified from the CAD file in the repository).
Will make some orifice plates just to test the rig, this test rig will not be used as an actual flow bench, I leave that build to my brother tha I'm helping with this project a true RatRig :D
![20241003_150454](https://github.com/user-attachments/assets/46117c9e-bafe-41f3-887f-0f9032ff32f4)
![20241003_150504](https://github.com/user-attachments/assets/89c0232b-2f88-40ee-8bea-961b6da834c2)
[75mm Dia MAF Mount v4 v7.zip](https://github.com/user-attachments/files/17250152/75mm.Dia.MAF.Mount.v4.v7.zip)
![maf-core-adapter-4-75mm-OD-Tube](https://github.com/user-attachments/assets/f55acb77-6d07-40f5-a03e-6c3d5731d83b)

**Comment by DeeEmm** [comment]
Looks great Stefan. Do you see any pulses in your system?

Also are you using any kind of velocity stack or rounding on transition of the port at the top?

**Reply by stefan63** [reply]
I'm not having any electronics up and running yet and that test bench are missing some parts to make it complete.
Design will probably change as I get the sesorboard up and running, I have all the components for it. and are working on a PCB light while waiting for a couple of your new boards :)

**Reply by stefan63** [reply]
Vill make some orifice plates for the top plate just to be able to do some basic tests.

**Reply by stefan63** [reply]
I have soldered up my ruff test PCB V 0.2 today it looks like sh... but at least no smoke came out at power up ;)
Will update my ESP tonight and if time allows I will make some initial tests of the board.
One MPXV7007 and one 6899 010A are installed, the bord allow one more MPXV to be installed.
MAF sensor will be connected at one of the screw terminals and the BME280 will be connected directly to the ESP with a short wire to keep it separate to avoid being heated at the bord installation location.   Doesnt look far as nice as the V2.3 board for sure ;) 
![pcb-2](https://github.com/user-attachments/assets/ddd9fc24-85e7-46c9-afba-681bdadab798)

**Reply by DeeEmm** [reply]
That looks great Stefan. Nice job. Well done.

**Reply by stefan63** [reply]
The board are now tested and some mistakes are corrected with jumpers, all sensors show changes when desktop tests performed that feels great :)
Will correct PCB layout later, will moste likly also incoporate jumpers in that version to since I intend to make a single sided PCB design for easy DIY and this board will not be fully featured as the V2.3 but I'm willing to share KiCad files for anyone interested.
Have to work 5x12.5hrs shift now so testing in my test bench rigg above has to wait :(  
Will report results and might be able to help better with the future development. 
Have a nice weekend /Stefan
![PCB-Working1](https://github.com/user-attachments/assets/ca635a47-b95f-46a7-9a46-0e684c5996a2)
![PCB-Working2](https://github.com/user-attachments/assets/323c6c0e-1dd0-42b9-a33e-446b35f35064)

**Reply by DeeEmm** [reply]
Awesome work Stefan.

A single sided design is the perfect solution for anyone wanting to make their own PCBs, and accessible for people using both PCB milling or etching.

There's lots of makers who love building down to this level, 'just because they can'.

Excellent stuff!!

**Reply by stefan63** [reply]
I have omited all extra cap filtering to make this as simple design as possible, further testing will reveal if I need to ad this later, if so I will siply solder some caps to the pads/tracks connecting the various modules. The resistor that can be spoted on the board are simly 0 ohm jumpers. The PCA9306 module thath I bought has already the enable circuit and the pullup resistors onboard that aslo made it possible to reduce the component count. Will post the KiCad files on this page when properly tested ;)
I also plan to include at least a SD card adapter to stor log files, and maybe........
![ADS1115](https://github.com/user-attachments/assets/da0c746c-248d-4e53-92e8-2e9e0d88a0c6)
![i2c-level](https://github.com/user-attachments/assets/9eae569d-4e14-48c7-b84b-e8e79d1540d1)

**Reply by DeeEmm** [reply]
I am interested to know how you get on without cap filtering. This is something that I have been pondering as a potential solution to the swings that @black-top has been experiencing.

**Reply by stefan63** [reply]
![flow1](https://github.com/user-attachments/assets/c091e19f-2127-43e0-a620-d73a08620f3f)
OK, I have done some initial tests today and I have some sanity check questions.
1. I use a CD as a test orifice and get a fluckturating value of ACFM like 9-25 with a median value of about 20 isch.
2. Using my Testo 405i I get a value 10X larger like 200cfm and it is quite steady (No ACFM setting on the Testo but this value is baro corrected ).
3. Is this as it should be, if not any hints why the valu of the MAF is 10X smaller than the Testo?   (When I flow over the MAF I can get the whole range 0-5V).
4. I'm thinking of putting a cap betwen MAF signal and Ground might help damping the flucturations a bit, will do some tests on that. I will also hook up my scope to se whats going on.

**Reply by DeeEmm** [reply]
Do you have to scale the value on the Testo for the size of duct being measured? Anemometers usually measure velocity, to convert to volumetric flow, you need to know the area of the duct. 

There is already a cap between the MAF and GND. (C7) This is 1uF. you can try increasing the value. It is only really there to help filter noise.

**Reply by stefan63** [reply]
Yes the Testo require a duct geometry and size to be entered and I have the same value 72mm dia in both the Testo and the DIY Flow bench UI. Will test some caps. I try to get as good result as possible without software filters, however the "Rolling Median" filter seams to do wounders and I get a stable reading +- 1 digit.  Will report my findings as I tamper.

**Reply by KirikauKiwi** [reply]
I have submitted a pull request that'll will hopefully resolve your scaling issue @stefan63

**Reply by KirikauKiwi** [reply]
@stefan63 what is DIYFB saying your MAF Flow is in KGH during this test?

**Reply by DeeEmm** [reply]
> Yes the Testo require a duct geometry and size to be entered and I have the same value 72mm dia in both the Testo and the DIY Flow bench UI. 

The MAF diameter is set in the MAF file, the value in the UI is ONLY if you are running  the MAF insert in a different sized pipe. For example you take the MAF cartridge out of the standard 72mm MAF housing and run it in a 100mm pipe. In this case you should enter 100mm in the GUI and the code will automatically rescale from 72mm to 100mm.

If you are running the standard MAF size, then the MAF Housing Diameter setting should be set to zero as this will bypass the scaling. It should also ignore the calculation if the GUI value equals the diameter from the MAF Data file

**Reply by KirikauKiwi** [reply]
@stefan63 what MAF sensor are you using?

**Reply by stefan63** [reply]
Using a spare MAF sensor that I bougt for my bad old Range Rover P38 -99 ;)
Bosch part number 0 280 218 010  It seams to be close to the AUDI one as I use that file.
I just took wath I had on hand and I guess It is kind of OK for functional tests. 
I might try to find a way to create a look up table along the journey...

**Reply by DeeEmm** [reply]
It is possible to create the lookup table using a known MAF with two MAFs in series. This is something that I planned to add to the code one day.

Should also be possible to do with your existing MAF and the flow meter but would be a more manual process.

One thing to watch out with the AUDI MAF is that it has a portion of the scale that is for reverse flow which the RR MAF will not have. ( up to 1v)

What is the diameter of the MAF?

**Reply by DeeEmm** [reply]
I suspect that it is a variant of the HF5 sensor, you should be able to check which one based on the diameter

<img width="507" alt="Screenshot 2024-11-11 at 6 34 58 pm" src="https://github.com/user-attachments/assets/4551166b-e181-4e7a-a6ef-24198064b9ce">

Perhaps the 0 280 218 008??

The following table gives you the output scaling

<img width="696" alt="Screenshot 2024-11-11 at 6 35 09 pm" src="https://github.com/user-attachments/assets/eab8b478-95f3-4ba8-922b-3882bacad2b7">

From this you can make a basic MAF Data file.

You can also try to interpolate the additional values in excel to make the full MAF Data file

**Reply by stefan63** [reply]
The original housing is 80mm but I use the blade only with a 3D printed adapter in a 72mm ID drain tube.
However the future sharp version that I will build for my brother will most likely use a know god MAF like the AUDI one in the list.
The reason I bult the test rig is mainly to be able to test things to be able to get some knowledge of how to route tubes and sensors to get a proper and as stable as possible readout. 3D printed a springloaded pressure release valve yesterday, will test it today to se if that will make some improvements, I can clearly hear the vac motors having an iregular running rpm, ned a way to get even flow from that box. The vacuum box already has a like 15 liters chamber between blowers and intake. I will report my findings in my Test rig buld thread when I get some results.

**Reply by stefan63** [reply]
Regarding the MAF I use, this is what I have found out.

Sensor with housing
BOSCH - Ref: 0 280 218 010

Sensor without housing
Manufacturer Part Number 0280217533 / 0280.217.533 / F00C.2G2.029

Some extra MAF info that I found online (I can't confirm that this is correct):

The Bosch part number on the sensor insert is F 00C 2G2 029

The 99-02 P38 uses the exact same part as a BMW M5

Some BMW M5 gurus use the sensor insert from a VW, F 00C 2G2 032
Connector is the same as 029 minus the latch (Can be solved with tie raps.

The MAF of 2000 Golf GL 2.0L Gas. VW PN: 06A 906 461A, Bosch PN 0 280 218 002, it has the Bosch OOC 2G2 032 sensor insert which can replace the 029 insert .

**Reply by DeeEmm** [reply]
Okay so it would most likely be the same as the 0280217531. From what I understand the HF5 insert is the same in all cases. 

I've made a quick MAFdata file based on the information in the Datasheet - it is in the branch BOSCH_0280218010

**Reply by stefan63** [reply]
I gave the new MAF setup a chance but the one I previously used BOSCH_0280218067 was acced thetually quite close.
If I simply cheated and reduced the duct size I could gett realy close to the values of my Testo anneometer.
I also tested my 3D printed pressure regulating valve and it seams to help a bit to mage the duct pressure less erratic.

**Reply by stefan63** [reply]
This is what I have found out and I guess that the Audi and Range Rover sensor "blades" are electricaly the same HFM7-R5.
Only mounting screw flange is different I guess that the ending 07C and 08P is mor like a physical variation number and the 0 280 218 part is the core specification. My initial tests with the  BOSCH_0280218010 with the BOSCH_0280218067.cpp file in a 72mm internal duct also make me beleive that this is the case.  I´m tempted to buy a 67 just to make shure haha..
It is very most impossible to get straigt answers from Bosch soooo....

HFM7-R5
AUDI BOSCH_0280218067
Independent aftermarket part (IAM)
Bosch 0 280 218 08P

HFM7-R5
Range Rover P38 Thor BOSCH_0280218010   
Independent aftermarket part (IAM)
Bosch 0 280 218 07C

**Reply by DeeEmm** [reply]
Ahhh so it's a HFM7 not HFM5. Pretty sure I've seen a HFM7 datasheet online. This should give you the correct transfer details. You can just substitute them into the MAF file I created.

What you are suggesting could well be true, in which case if the values are the same then you can use the details from the AUDI MAF file. However there is a discrepancy in that any value <1v is reverse flow on the AUDI MAF, which I believe is unique to the RS4.

**Comment by KirikauKiwi** [comment]
@DeeEmm just spotted this, I think it explains the order of magnitude problem @stefan is seeing.
```
		// Calculate the 'velocity' for the original pipe area
		mafVelocity = flowRateKGH / PI * pow((status.mafDiameter / 2), 2);

		// scale the result with the new pipe area and convert back to mass flow
		transposedflowRateKGH = mafVelocity * PI * pow((config.maf_housing_diameter / 2), 2);
```
The MAF diameter is picked up in mm from the sensors.cpp and from the MAF Housing Diameter on the Configuration GUI page, I cannot see it being converted to meters which is necessary for the above conversions.

Can you check this please.

**Reply by DeeEmm** [reply]
I believe this is correct.  I am unsure of ...

> being converted to meters which is necessary for the above conversions.

The formula is just the area of a circle (PI * R^2). The dimensions used are unimportant as long as you use the same dimensions for both parts of the equation.

We take the area of the original MAF...
```
PI * pow((status.mafDiameter / 2), 2
```

And the area of the new housing
```
PI * pow((config.maf_housing_diameter / 2), 2);
```

And use it to scale the flow

The method is in the comments above the formula...

```
// Q = A*V
// V = Q/A
// A = PI*r^2
// where Q = flow | A = area | V = velocity
```

**Reply by KirikauKiwi** [reply]
Yes you are in fact correct in this example where the two calculations are used together, they will cancel each other out since you are dividing in one equation and multiplying in the other equation.

I was not sure if they were used anywhere else in the code. If they are used singularly elsewhere they will yield a result that is out by a factor about 6 e.g. if the diameter was 82mm the r<sup>2</sup>would be 41<sup>2</sup>=1681mm<sup>2</sup> and 0.041<sup>2</sup>=0.001681m<sup>2</sup>

Build 24110802
Version V.2.0-RC.8
There is something else at play then. Below I have entered 72 in to the MAF Housing diameter on the config page and get this.
<img width="717" alt="Screenshot 2024-11-10 at 15 19 16" src="https://github.com/user-attachments/assets/bbc424e7-18a7-43d9-bf1b-ba8286ef50d1">

From this Mass Flow
<img width="717" alt="Screenshot 2024-11-10 at 15 20 01" src="https://github.com/user-attachments/assets/5156c6b1-9703-4e13-a0c4-7b3e3fecaa13">

I am seeing the same thing in the SCFM-Conversion-#208 too.

@DeeEmm can you check to see if the same is happening for you?

**Reply by KirikauKiwi** [reply]
Calculating this manually for the MAF I have defined "Bosch" 82mm and using 72mm for the MAF Housing Diameter and using the MAF flow from above of 363.6 kg/h I get a transposedflowRateKGH = 280.324569

I have meet the conditions and it appears the calculation is being performed as per the code, because if I change the MAF housing diameter back to zero I get the original MAF Flow back.
Where does the transposedflowRateKGH output then get used in the code? to be displayed or used for other calculations if needed.

```
// Now that we have a converted flow value we can translate it for different housing diameters
	if (config.maf_housing_diameter > 0 && status.mafDiameter > 0 && config.maf_housing_diameter != status.mafDiameter) { 
		// We are running a custom MAF Housing, lets translate the flow rates to the new diameter

		// NOTE this transfer function is actually for volumetric flow but we can assume that the mass to volumetric conversion is ratiometric at any given instant in time.
		// This is because any environmental influences apply equally to both parts of the equation and cancel each other out. 
		// So we can reduce the mass flow conversion to the same simple ratio that we use for volumetric flow. 
		// In short, the relationship between flow and pipe area applies equally to both mass flow and volumetric flow for a given instant in time.

		// Q = A*V
		// V = Q/A
		// A = PI*r^2
		// where Q = flow | A = area | V = velocity

		// Calculate the 'velocity' for the original pipe area
		mafVelocity = flowRateKGH / PI * pow((status.mafDiameter / 2), 2);

		// scale the result with the new pipe area and convert back to mass flow
		transposedflowRateKGH = mafVelocity * PI * pow((config.maf_housing_diameter / 2), 2);

		return transposedflowRateKGH;

	} else { // lets send the standard MAF data

		return flowRateKGH;
```

**Reply by DeeEmm** [reply]
>  Where does the transposedflowRateKGH output then get used in the code? to be displayed or used for other calculations if needed.
> 

It is returned from the class method Sensors::getMafFlow(int units) as shown in the code you quote above...

```
return transposedflowRateKGH;
```

Which is called from hardware.cpp

```
mafFlowRateCFM = _calculations.convertFlow(_sensors.getMafFlow());
```

So it is part of the normal code flow. 

The same method is also called in the API...

```
      case 'f': // Get measured Mass Flow 'F123.45\r\n'       
          // snprintf(apiResponse, API_RESPONSE_LENGTH, "f%s%f", config.api_delim , sensorVal.FlowKGH);
          snprintf(apiResponse, API_RESPONSE_LENGTH, "f%s%f", config.api_delim , _sensors.getMafFlow());
      break;
```

So interrogating the value in the serial monitor will show you exactly what is returned from the getMafFlow() method.

**Reply by KirikauKiwi** [reply]
I am starting to work out the flow of the code!!!! it is a bit like trying to read a foreign language at times.

With the most current build 24111102, I am still getting wacky numbers on the GUI when I enter a value in the MAF Housing diameter on the config page.

They appear at least in the ball park and correct order of magnitude when I divide the result by one million., see below.
```
// Calculate the 'velocity' for the original pipe area
		mafVelocity = flowRateKGH  / PI * pow(status.mafDiameter / 2, 2);

		// scale the result with the new pipe area and convert back to mass flow
		transposedflowRateKGH = mafVelocity * PI * pow(config.maf_housing_diameter / 2, 2) / 1000000;

		return transposedflowRateKGH;
```
This is what I get on the GUI,
<img width="717" alt="Screenshot 2024-11-10 at 23 09 26" src="https://github.com/user-attachments/assets/58f5715d-97fa-4fc2-ac6c-d55d59e95352">

Again without the divide by one million.
```
		// Calculate the 'velocity' for the original pipe area
		mafVelocity = flowRateKGH  / PI * pow(status.mafDiameter / 2, 2);

		// scale the result with the new pipe area and convert back to mass flow
		transposedflowRateKGH = mafVelocity * PI * pow(config.maf_housing_diameter / 2, 2);

		return transposedflowRateKGH;
```
This is what I get on the GUI,
<img width="717" alt="Screenshot 2024-11-10 at 23 12 52" src="https://github.com/user-attachments/assets/a92b78ed-19a7-4f88-8f29-0f584cf244bd">

**Reply by DeeEmm** [reply]
So I'm using the BOSCH_0280218067.cpp which has a diameter of 82mm.

If I enter 0 or 82, the results are as expected. But if I change that to 90 then I see a similar outcome...

<img width="353" alt="Screenshot 2024-11-11 at 10 55 53 am" src="https://github.com/user-attachments/assets/cb4c8d6c-b03a-484b-ade0-12331a7ff2ab">

Will take a look later this eve.

**Reply by DeeEmm** [reply]
Created issue #217 for this.

**Comment by stefan63** [comment]
Nice to know, on Build:24110901 the MAF value now corresponds to the value I gett from my Testo 405.
The Testo isn't much more stable than the reading from the MAF sensor and are quite sensitive to angle against air stream so I guess that is something we have to live with.

**Comment by stefan63** [comment]
OK, had some small tubing at home and mocked up a quick and dirty U-pitot probe will test it later and se if it possibly works.
4mm soft copper tube (3mm id) and a stainles steel tube 2mm YD 1.7mm id, 2x1mm holes for static intake. Have do idea if those tubes are to narrow but air flows where is should, so tests might tell :D
I used a bicykle brake wire to support the steel tube and bent them both together, for MK2 I will fabricate a bending guide :0)
![20241122_135618](https://github.com/user-attachments/assets/d5b33ff6-42d1-480c-be55-ff7f1282ac75)
![20241122_125211](https://github.com/user-attachments/assets/b661e6de-2bd2-4ab4-861e-49a213a8a58a)
![20241122_125306](https://github.com/user-attachments/assets/b6626062-401e-4fa8-8ac0-5b9cb19310c6)

**Comment by DeeEmm** [comment]
Awesome work Stefan. Great idea on using the cable as support. 

Keen to see your results. 👍

**Comment by stefan63** [comment]
Found a suitable (as I believe) 63mm manual operated ball valve on Printables that I ave modified to become a NEMA17 stepper motor controlled valve. I made it to be mounted on a box to act fine tuning bleed actions or if you like you can change it to be duct mounted. Not tested yet but I´m confident that it will work. 
Download here: https://www.printables.com/model/1098412-stepper-motor-controlled-63mm-ball-valve
![v1](https://github.com/user-attachments/assets/84de6691-329d-43e2-ab24-59548c33b528)
![v2](https://github.com/user-attachments/assets/cc8090e5-ec3f-4ee4-8b2f-4ef6b84e83f3)
![v3](https://github.com/user-attachments/assets/fdce71ea-8b91-4cb6-b90b-73dd6dcbf197)
![v4](https://github.com/user-attachments/assets/01990bee-af96-4c9c-bb60-657cf2176ace)
![V6](https://github.com/user-attachments/assets/e0f6ec5a-70b7-4c59-bea1-bc800c0d761d)

Based on: https://www.printables.com/model/137795-63mm-2-12-ball-valve-type-non-clog-blast-gate-for

**Reply by KirikauKiwi** [reply]
Is there any o-ring or other type of gasket inside to help with the sealing?

**Reply by stefan63** [reply]
> Is there any o-ring or other type of gasket inside to help with the sealing?

Nope, and I dont think it is needed since it is a bleed valve and you should always deliver an exess of perssure to be regulated.
If you are thinking of leak test, just simply block it temporary with some tape or remove it completely and cover the opening.
I feel that the leek test is more like a build test, when done you are kind of done ;)

**Reply by stefan63** [reply]
I also have a disk type valve and a slit valve on the drawing board, will post later.
